Packs/GenericSQL/Integrations/GenericSQL/GenericSQL.py

```diff
@@ -165,8 +167,16 @@ def _create_engine_and_connect(self) -> sqlalchemy.engine.base.Connection:
             cache_key = self._get_cache_string(str(db_url), ssl_connection)
             engine = cache.get(cache_key, None)
             if engine is None:  # (first time or expired) need to initialize
+
+                engines = getattr(sqlalchemy, ENGINES, {})
+
+                if cache_key in engines:
+                    engines[cache_key].dispose()
```

Comment:
do we want to dispose the engine on each command, or only if it's beyond its ttl?

Reply:
only if it's beyond its ttl

Reply:
where in the code?